### PR TITLE
Add expo-in-app-purchases to bare-expo and native-component-list

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Step 1 to deleting apps/expo-payments

# How

Move expo-payments demo to bare-expo and native-component-list

# Test Plan

Build bare-expo (ios, Android), open IAP demo.

### CODE REVIEW

File: apps/bare-expo/ios/Pods/Headers/Private/EXSQLite/EXSQLite.h [hunk 1]
```diff
@@ -1 +1 @@
-../../../../../../../packages/expo-sqlite/ios/EXSQLite/EXSQLite.h
\ No newline at end of file
+../../../../../../../packages/expo-sqlite/ios/EXSQLite/EXSQLite.h
```
Comment (sjchmiela) [inline apps/bare-expo/ios/Pods/Headers/Private/EXSQLite/EXSQLite.h]: There's no difference and the diff is still showing. 🤔 Maybe squashing offending commits would help remove this entry?

File: apps/native-component-list/src/screens/InAppPurchases/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Expo Payments
+
+This is an ejected Expo app that can be used to test the `expo-in-app-purchases` module.
+
+It has a bundle ID of `dev.expo.Payments` and an application ID of `com.expo.payments`.
```
Comment (sjchmiela) [inline apps/native-component-list/src/screens/InAppPurchases/README.md]: `bare-expo` uses an application ID of `dev.expo.payments`. Do you think we should change the respective configuration in the Google Play Console?

File: apps/native-component-list/src/screens/InAppPurchases/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Expo Payments
+
+This is an ejected Expo app that can be used to test the `expo-in-app-purchases` module.
```
Comment (sjchmiela) [inline apps/native-component-list/src/screens/InAppPurchases/README.md]:
```suggestion
This screen may not be usable outside of the `bare-expo` project, which is preconfigured for supporting in-app purchases.
```

**Review by sjchmiela — COMMENTED** [review]
👍 👍 👍

File: apps/native-component-list/src/screens/InAppPurchases/README.md (hunk 2 above)
Comment (EvanBacon) [reply apps/native-component-list/src/screens/InAppPurchases/README.md]: Might make sense to just create a few more payments.

**Pull request merged**